modules/Link.js

```diff
@@ -62,6 +77,60 @@ const Link = React.createClass({
     }
   },
 
+  getInitialState() {
+    return {
+      isActive: isLinkActive(this.context.router, this.props)
+    }
+  },
+
+  componentWillMount() {
+    this._isActive = this.state.isActive
+  },
+
+  componentDidMount() {
+    const { router } = this.context
+    if (router) {
+      this._unlisten = router.listen(() => {
```

Comment:
This PR switches `router.listen` to be implemented by the transition manager so it reports the tree about to be rendered.